modules/sdk-coin-atom/src/lib/transaction.ts

```diff
@@ -0,0 +1,193 @@
+   *
+   * @param {TransactionType} transactionType The transaction type to be set.
+   */
+  transactionType(transactionType: TransactionType): void {
+    this._type = transactionType;
+  }
+
+  /**
+   * Sets this transaction payload
+   *
+   * @param rawTransaction
+   */
+  fromRawTransaction(rawTransaction: string): void {
+    try {
+      this._atomTransaction = Transaction.deserializeAtomTransaction(rawTransaction);
+      this._type = TransactionType.Send;
+    } catch (e) {
+      throw e;
+    }
+  }
+
+  serialize(): string {
+    // TODO BG-67032 - if possible use Stargate Client builder to build [and sign, if applicable] tx using data types from this tx class
+    // Needs research: how to get stargate client builder without feeding an OfflineSigner private key
+    throw new Error('not yet implemented');
+  }
+
+  static deserializeAtomTransaction(txHex: string): AtomTransaction {
+    const decodedTx = decodeTxRaw(fromBase64(txHex));
+    const registry = new Registry(defaultRegistryTypes);
+    // TODO - support many messages decoding, right now we are just taking one message for send tx type
+    const encodedMessages = decodedTx.body.messages as EncodeObject[];
+    const decodedMessage = registry.decode(encodedMessages[0]); // TODO - do we need this?
+    let type: AtomTransactionType;
+    if (encodedMessages[0].typeUrl === sendMsgType) {
+      type = AtomTransactionType.Pay;
+    } else {
+      throw new Error('Transaction type not supported: ' + encodedMessages[0].typeUrl);
+    }
+    const sendMessageData: MessageData = {
+      typeUrl: decodedMessage.typeUrl,
+      value: {
+        fromAddress: decodedMessage.fromAddress,
+        toAddress: decodedMessage.toAddress,
+        amount: decodedMessage.amount,
+      },
+    };
+    const sequence = Number(decodedTx.authInfo.signerInfos[0].sequence);
+    const explicitSignerData: SignerData = {
+      accountNumber: 100, // TODO - get this from tx data somehow
+      sequence,
+      chainId: thetaTestnetChainId, // TODO - figure out if we can get this from tx data
+    };
+    const gasBudget: GasFeeLimitData = {
+      amount: decodedTx.authInfo.fee?.amount as Coin[],
+      gas: Number(decodedTx.authInfo.fee?.gasLimit),
+    };
+    return {
+      type,
+      sendMessage: sendMessageData,
+      signerAddress: sendMessageData.value.fromAddress,
+      gasBudget: gasBudget,
+      explicitSignerData,
+    };
+  }
+
+  /**
+   * Returns a complete explanation for a transfer transaction
+   * Currently only supports one message per transfer.
+   * @param {TxData} json The transaction data in json format
+   * @param {TransactionExplanation} explanationResult The transaction explanation to be completed
+   * @returns {TransactionExplanation}
+   */
+  explainTransferTransaction(json: TxData, explanationResult: TransactionExplanation): TransactionExplanation {
+    const outputAmount = this.atomTransaction.sendMessage.value.amount[0].amount;
```

Comment:
More of a question for my understanding since I'm new to typescript:

I see that we've initialized a class variable called `_atomTransaction` with a leading underscore to indicate that it's private. However, in line 180 and 183, I see that underscore is not present. So my question is whether this is interchangeable or a typo ? If it's interchangeable, could we stick to one way of referencing the class variable (either with _ or the regular way) ?

Reply:
Ah gotcha, I read the `get atomTransaction()` as `getAtomTransaction()`. Now it makes sense